tools/clang/tools/dxcompiler/dxcvalidator.cpp

```diff
@@ -122,6 +131,34 @@ HRESULT STDMETHODCALLTYPE DxcValidator::Validate(
   return ValidateWithOptModules(pShader, Flags, nullptr, nullptr, ppResult);
 }
 
+HRESULT STDMETHODCALLTYPE DxcValidator::ValidateWithDebug(
+  _In_ IDxcBlob *pShader,                       // Shader to validate.
+  _In_ UINT32 Flags,                            // Validation flags.
+  _In_ DxcBuffer *pDebugModule,                 // Debug module to provide line numbers
+  _COM_Outptr_ IDxcOperationResult **ppResult   // Validation output status, buffer, and errors
+)
+{
+  DxcThreadMalloc TM(m_pMalloc);
+  if (pShader == nullptr || ppResult == nullptr || Flags & ~DxcValidatorFlags_ValidMask)
+    return E_INVALIDARG;
+  if ((Flags & DxcValidatorFlags_ModuleOnly) && (Flags & (DxcValidatorFlags_InPlaceEdit | DxcValidatorFlags_RootSignatureOnly)))
+    return E_INVALIDARG;
+  if (pDebugModule && (pDebugModule->Ptr == nullptr || pDebugModule->Size == 0))
+    return E_INVALIDARG;
+
+  std::unique_ptr<llvm::Module> pM;
+  std::unique_ptr<llvm::MemoryBuffer> pMemBuf;
+  LLVMContext ctx;
+  if (pDebugModule) {
+    pMemBuf = llvm::MemoryBuffer::getMemBuffer(StringRef((char *)pDebugModule->Ptr, pDebugModule->Size), "", false);
```

Comment:
As the clang error implies, this will need to be cast to `(const char*)` with that change, it builds with clang

Reply:
Yes, and that reminds me about other issues which make it necessary to refactor this a bit more in any case.